src/v/rpc/types.h

```diff
@@ -233,8 +234,10 @@ struct method {
 /// \brief used in returned types for client::send_typed() calls
 template<typename T>
 struct client_context {
-    explicit client_context(header h)
-      : hdr(std::move(h)) {}
+    client_context(header h, T data)
```

Comment:
Why is data useful here

Reply:
Ah I see it. It seems like 2 different patches. One is for error handling and the other is an optimization 

is a nit. But in the future split up the prs

Reply:
those changes are in separate commits(i.e. small ctor refactoring), do you think it is enough or would you prefer separate PR  ?

Reply:
is good :)